File: BundlePassingPractice/app/src/main/java/com/example/bundlepassingpractice/SecondActivity.java
package com.example.bundlepassingpractice;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    TextView textStudentInfo;
    TextView textRollNoInfo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        //Finding IDs
        textStudentInfo = findViewById(R.id.textStudentNameInfo);
        textRollNoInfo = findViewById(R.id.textRollNoInfo);

        //getting intent from Main Activity
        Intent fromMainActivity = getIntent();

        String actionBarName = fromMainActivity.getStringExtra("actionBarName");
        String studentName = fromMainActivity.getStringExtra("studentName");
        int rollNo = fromMainActivity.getIntExtra("rollNo",0);


        //getting intent from Main Activity without setting them into Intent datatype
//        String actionBarName = getIntent().getStringExtra("actionBarName");
//        String studentName = fromMainActivity.getStringExtra("studentName");
//        int rollNo = fromMainActivity.getIntExtra("rollNo",0);

        //displaying Details
        textStudentInfo.setText("Student Name: "+studentName);
        textRollNoInfo.setText("Roll No: "+rollNo);

        //change heading of toolBar
        getSupportActionBar().setTitle(actionBarName);


        //directly setting details in TextView
//        textStudentInfo.setText("StudentName: "+getIntent().getStringExtra("studentName"));
//        textRollNoInfo.setText("Roll No: "+getIntent().getIntExtra("rollNo",0));
//        getSupportActionBar().setTitle(getIntent().getStringExtra("actionBarName"));

    }
}

File: BundlePassingKotlin2/app/src/main/java/com/himanshu/bundlepassingkotlin2/SecondActivity.kt
package com.himanshu.bundlepassingkotlin2

import android.os.Build
import android.os.Bundle
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
class SecondActivity : AppCompatActivity() {


    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val textView: TextView = findViewById(R.id.textView)

        val messageToDisplay = intent.getSerializableExtra("EXTRA_PERSON", Person::class.java)

        textView.text = messageToDisplay.toString()




    }


}